numpy/f2py/crackfortran.py

```diff
@@ -84,7 +84,7 @@
                        'optional','required', etc)
      K = D['kindselector'] = {['*','kind']} (only if D['typespec'] =
                          'complex' | 'integer' | 'logical' | 'real' )
-     C = D['charselector'] = {['*','len','kind']}
+     C = D['charselector'] = {['*','len','kind','f2py_len']}
```

Comment:
Should the `f2py_*` directives be handled separately from the Fortran standard selectors? (not very important)

Reply:
Yes, all expressions within `f2py` directives are expected to be valid C expressions and are evaluated on the C side. However, there may be occasions where the expressions may be evaluated at the Fortran side as well (e.g. in generated Fortran wrapper codes) but the current f2py is ignorant of this and https://github.com/numpy/numpy/pull/19805 is a step toward translating f2py-directive expressions to appropriate language target.